Training24/Test.c

```diff
@@ -1,19 +1,187 @@
+         printf ("Invalid choice. Please enter 1, 2, or 3.\n");
+         continue;
+      }
+      char* start = option;
+      while (isspace (*start)) start++;
+      switch (*start) {
+      case '1':
+         printf (CYAN "\n\t   -------------  TESTING PHRASE PALINDROME  -------------\n" RESET);
+         TestUserPhrase (); break;
+      case '2':
+         printf (CYAN "\n\t   -------------  TESTING INTEGER PALINDROME  -------------\n" RESET);
+         TestUserInteger (); break;
+      case '3':
+         printf ("\nExiting program.\n"); return 0;
+      default:
+         printf ("Invalid choice. Please enter 1, 2, or 3.\n"); break;
+      }
+   }
+   return 0;
+}
+
+enum HeaderType { Phrase, Integer };
+void PrintTableHeader (enum HeaderType type) {
+   switch (type) {
+   case Phrase:
+      printf ("\n\t+----------------------+-------------------+-------------------+--------+\n"
+         "\t|  PHRASE - TEST CASE  |  EXPECTED OUTPUT  |   ACTUAL OUTPUT   | RESULT |\n"
+         "\t+----------------------+-------------------+-------------------+--------+\n");
+      break;
+   case Integer:
+      printf ("\n\t+---------------------+--------------------------+-------------------+-------------------+--------+\n"
+         "\t| INTEGER - TEST CASE |     REVERSED INTEGER     |  EXPECTED OUTPUT  |   ACTUAL OUTPUT   | RESULT |\n"
+         "\t+---------------------+--------------------------+-------------------+-------------------+--------+\n");
+      break;
+   default:
+      printf ("\tInvalid header type.\n"); break;
+   }
+}
+
+void PhraseTableRow (const char* input, const char* expectedOutput, const char* actualOutput, const char* result) {
+   printf ("\t| %-20s | %-17s | %-17s | %-15s |\n", input, expectedOutput, actualOutput, result);
+   printf ("\t+----------------------+-------------------+-------------------+--------+\n");
+}
+
+void IntegerTableRow (int number, const char* outputStr, const char* actualOutput, const char* expectedOutput) {
+   const char* resultStr = (strcmp (actualOutput, expectedOutput) == 0) ? GREEN "PASS" RESET : RED "FAIL" RESET;
+   printf ("\t| %-19d | %-24s | %-17s | %-17s | %-15s |\n", number, outputStr, actualOutput, expectedOutput, resultStr);
+   printf ("\t+---------------------+--------------------------+-------------------+-------------------+--------+\n");
+}
+
+void TestPhrasePalindromes () {
+   PrintTableHeader (Phrase);
+   const char* phrases[] = { "tacOcAT", "hello", "D", "Don't nod", "Race!Car", " ", "22 / 02 / 2022",
+      "12345678987654321", "0123456789" },
+      * expectedResults[] = { "Palindrome", "Not a Palindrome", "Palindrome", "Palindrome",
+      "Palindrome", "Palindrome", "Palindrome", "Palindrome", "Not a Palindrome" };
+   for (int i = 0; i < sizeof (phrases) / sizeof (phrases[0]); i++) {
+      int isPhrasePalindrome = CheckPhrasePalindrome (phrases[i]);
+      const char* actualResult = isPhrasePalindrome ? "Palindrome" : "Not a Palindrome",
```

Comment:
Since you are only using the value in one place
```suggestion
      const char* actualResult = CheckPhrasePalindrome (phrases[i])? "Palindrome" : "Not a Palindrome",
```